### ISSUE TITLE
`update` gives a different result than `elm/core`

### ISSUE THREAD

**Issue by miniBill** [matzko/elm-opaque-dict#2]
```elm
> import Dict
> let a = Dict.singleton 0 0 in Dict.update 0 (\_ -> Nothing) a
Dict.fromList [] : Dict.Dict number number1
> let a = Dict.singleton 0 0 in Dict.update 0 (\_ -> Nothing) a
Dict.fromList [] : Dict.Dict number number1
```

When using `update`, mapping to `Nothing` means to remove the element, whereas in this package it leaves it untouched.

Fixing this would be a breaking change because it changes the behavior, but not a breaking API change.

I see three approaches:
1) document the difference
2) fix the implementation (from `withDefault ... >> Tuple.pair ... >> Just` to `Maybe.map (Tuple.pair ...)`), put a warning somewhere, hope users read it
3) add a dummy export, publish a minor version, then fix as above while removing the dummy export to force a major version (this still requires a warning but should avoid surprises, users are _supposed_ to check what a major version bump entails

**Comment by Janiczek** [comment]
Note that apart from this "removal" bug, there's also a bug with the "creation", ie. the `Nothing -> Just` case:

```elm
update k (\_ -> Just v) empty
```
* elm/core `Dict` will return a dict with the `(k, v)` pair
* `OpaqueDict` will return an empty dict

<img width="544" alt="Screenshot 2022-12-17 at 0 19 43" src="https://user-images.githubusercontent.com/149425/208204148-431c8f12-b036-4904-819b-2f2dcd0f61f9.png">

**Comment by matzko** [comment]
Fixed both in [2.0.0](https://github.com/matzko/elm-opaque-dict/blob/main/CHANGELOG.md#200---2024-04-09) with the breaking change to make compatible with `Dict.update`.  Thanks!
